File: src/main/kotlin/io/paulocosta/twitbooks/goodreads/GoodreadsBook.java
package io.paulocosta.twitbooks.goodreads;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Root(strict = false)
public class GoodreadsBook {

    public GoodreadsBook() {}

    @Element(name = "title")
    private String title;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}

File: backend/src/main/kotlin/io/paulocosta/twitbooks/books/provider/goodreads/GoodreadsBook.kt
package io.paulocosta.twitbooks.books.provider.goodreads;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Root(strict = false)
data class GoodreadsBook @JvmOverloads constructor(

    @field:Element(name = "id")
    @param:Element(name = "id")
    var id: Long? = null,

    @field:Element(name = "title")
    @param:Element(name = "title")
    var title: String? = null,

    @field:Element(name = "small_image_url")
    @param:Element(name = "small_image_url")
    var smallImageUrl: String? = null,

    @field:Element(name = "image_url")
    @param:Element(name = "image_url")
    var imageUrl: String? = null

)
